### PR TITLE
Delete 'transfigure' tool

### PR DESCRIPTION
**Pull request by chases2**
Transfigure is a deprecated approach to adding configurations to TestGrid; it has been replaced with Config Merging.

The [last PR](https://github.com/kubernetes/test-infra/pull/25295) the Transfigure tool proposed in k8s/test-infra was in April of last year, and was closed when that team migrated to Config Merging. I don't think anyone is using this tool, and it should be deleted so it does not need to be maintained.

Testing: `make test` works locally for this PR

### CODE REVIEW

**Comment by michelle192837** [comment]
I think https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md#prow-job-configuration also needs an update? Otherwise, LGTM!

**Comment by chases2** [comment]
> I think https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md#prow-job-configuration also needs an update? Otherwise, LGTM!

Good point! Updated the README to reflect.
/hold cancel

**Review by michelle192837 — APPROVED** [review]

**Pull request merged**
